cloud: update roadmap description (#10587)
pingcap/docs: tidb-cloud/tidb-cloud-roadmap.md

<table>
<thead>
  <tr>
    <th>Domain</th>
    <th>Feature</th>
    <th>Description</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>Key rotation</td>
    <td>Support key rotation on TiDB clusters for AWS.</td>
    <td>Support key rotation on TiDB clusters to improve the security of encrypted data.</td>
  </tr>
  <tr>
    <td>Key management</td>
    <td>Support making your own key manageable (BYOK from AWS).</td>

>     <td>Allow customers to use their own data encryption keys on AWS.</td>

Reviewer: ```suggestion
    <td>Allow you to use your own data encryption keys on AWS.</td>
```

  </tr>
   <tr>
    <td>Audit logging</td>

>     <td>Enhance the database audit logging feature.</td>

Reviewer: ```suggestion
    <td>Enhance the database audit logging.</td>
```
